app/src/main/java/ch/epfl/sdp/peakar/database/Database.java

```diff
@@ -54,4 +61,20 @@ public static Database getInstance() {
     public DatabaseReference getReference() {
         return reference;
     }
+
+    /**
+     * Enable the DB offline mode.
+     * Requests for which data is cached can still be answered.
+     * Other requests will be enqueued until online mode is set.
+     */
+    public void setOfflineMode() {
+        FirebaseDatabase.getInstance(DATABASE_ADDRESS).goOffline();
+    }

```

Comment:
I have not understood how this is called, shouldn't there be a `SharedPreferenceListener`?

Reply:
Indeed it is not called, I've just implemented the method to be called in order to disconnect the DB.